dali/operators/image/color/color_twist.cc

```diff
@@ -164,20 +181,30 @@ void ColorTwistCpu::RunImpl(workspace_t<CPUBackend> &ws) {
   output.SetLayout(input.GetLayout());
   auto &tp = ws.GetThreadPool();
   TYPE_SWITCH(input.type(), type2id, InputType, (uint8_t, int16_t, int32_t, float, float16), (
-      TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float, float16), (
-          {
-              using Kernel = TheKernel<OutputType, InputType>;
-              for (int i = 0; i < input.shape().num_samples(); i++) {
-                tp.AddWork([&, i](int thread_id) {
-                  kernels::KernelContext ctx;
-                  auto tvin = view<const InputType, 3>(input[i]);
-                  auto tvout = view<OutputType, 3>(output[i]);
-                  kernel_manager_.Run<Kernel>(ws.thread_idx(), i, ctx, tvout, tvin,
-                                              tmatrices_[i], toffsets_[i]);
-                }, out_shape.tensor_size(i));
-              }
+    TYPE_SWITCH(output_type_, type2id, OutputType, (uint8_t, int16_t, int32_t, float, float16), (
+      {
+        using Kernel = TheKernel<OutputType, InputType>;
```

Comment:
I see that most of the code is duplicated in ColorTwist family and in BrightnessContrast family. I see the main difference is that one uses a linear transformation kernel and the other uses MultiplyAdd kernel. Is there a good reason not to unify to a single implementation?

Reply:
🤷‍♂️